apimodels/task_log.go

```diff
@@ -35,9 +51,107 @@ type TaskLog struct {
 	Messages     []LogMessage `json:"m"`
 }
 
-// BuildloggerLogLine represents a cedar buildlogger log line, with the
-// severity extracted.
-type BuildloggerLogLine struct {
-	Message  string `bson:"m" json:"m"`
-	Severity string `bson:"s" json:"s"`
+func GetSeverityMapping(s int) string {
+	switch {
+	case s >= int(level.Error):
+		return LogErrorPrefix
+	case s >= int(level.Warning):
+		return LogWarnPrefix
+	case s >= int(level.Info):
+		return LogInfoPrefix
+	case s < int(level.Info):
+		return LogDebugPrefix
+	default:
+		return LogInfoPrefix
+	}
+}
+
+func GetBuildloggerLogs(ctx context.Context, buildloggerBaseURL, taskId, logType string, tail, execution int) (io.ReadCloser, error) {
```

Comment:
all public funcs should have a comment explaining the what / why

Reply:
Our convention for go code is to not require docstrings on public functions, though if you feel that it'll help, adding them is encouraged. I agree that these could use some comments explaining why they return a reader rather than an array of data, but this was just moved from 1 file to another so I think it's fine to not add in this PR

Reply:
should the linter be changed then? right now it gives the yellow squiggle line and complains of a warning "should have comment"

Reply:
I think that's the linter in your editor - I believe the default go extension for VScode has this check. Lint is passing in this PR